### DISCUSSION TITLE
KEDA Autoscaling for selenium grid || FailedGetExternalMetric  the HPA was unable to compute the replica count || unable to get external metric prod/s0-seleniumgrid-MicrosoftEdge

### DISCUSSION THREAD

**Discussion by vinuvk11** [kedacore/keda#5682] [Q&A / Need Help] [answered]
kubectl get hpa -n prod
NAME                          REFERENCE            TARGETS             MINPODS   MAXPODS   REPLICAS   AGE
keda-hpa-selenium-grid-edge   Deployment/browser   <unknown>/1 (avg)   5         50        5          44h

kubectl describe hpa keda-hpa-selenium-grid-edge -n prod

Metrics:                                                   ( current / target )
  "s0-seleniumgrid-MicrosoftEdge" (target average value):  <unknown> / 1
Min replicas:                                              5
Max replicas:                                              50
Deployment pods:                                           5 current / 0 desired
Conditions:
  Type           Status  Reason                   Message
  ----           ------  ------                   -------
  AbleToScale    True    SucceededGetScale        the HPA controller was able to get the target's current scale
  ScalingActive  False   FailedGetExternalMetric  the HPA was unable to compute the replica count: unable to get external metric prod/s0-seleniumgrid-MicrosoftEdge/&LabelSelector{MatchLabels:map[string]string{scaledobject.keda.sh/name: selenium-grid-edge,},MatchExpressions:[]LabelSelectorRequirement{},}: unable to fetch metrics from external metrics API: the server is currently unable to handle the request (get s0-seleniumgrid-MicrosoftEdge.external.metrics.k8s.io)
Events:
  Type     Reason                   Age                    From                       Message
  ----     ------                   ----                   ----                       -------
  Warning  FailedGetExternalMetric  112s (x5094 over 21h)  horizontal-pod-autoscaler  unable to get external metric prod/s0-seleniumgrid-MicrosoftEdge/&LabelSelector{MatchLabels:map[string]string{scaledobject.keda.sh/name: selenium-grid-edge,},MatchExpressions:[]LabelSelectorRequirement{},}: unable to fetch metrics from external metrics API: the server is currently unable to handle the request (get s0-seleniumgrid-MicrosoftEdge.external.metrics.k8s.io)

Error is unable to auto scale the selenium grid browser sessions

#$ cat keda_selenium.yaml
apiVersion: keda.sh/v1alpha1
kind: ScaledObject
metadata:
  name: selenium-grid-edge
  namespace: prod
  labels:
    deploymentName: browser
spec:
  minReplicaCount: 5
  maxReplicaCount: 50
  pollingInterval: 3
  cooldownPeriod:  30
  scaleTargetRef:
    name: browser
  triggers:
    - type: selenium-grid
      metadata:
        url: 'http://selenium-hub.prod:4444/graphql'
        browserName: 'MicrosoftEdge'
        unsafeSsl: 'true'
        sessionBrowserName: 'msedge'
        #maxQueuedSessions: '5'

**Comment by zroubalik** [comment, marked as answer]
I think that everything is correct, it is just old event displayed in HPA, I can see there:
```
AbleToScale True SucceededGetScale the HPA controller was able to get the target's current scale
```

**Comment by vinuvk11** [comment]
thanks for response but my autoscaling is not happening still  i cloud  feel error in two areas

**Comment by vinuvk11** [comment]
HPA targets are shown unknown 
<img width="740" alt="image" src="https://github.com/kedacore/keda/assets/165902584/2a99bf28-76fd-4eb3-a538-e15c82989f96">

**Comment by vinuvk11** [comment]
while describing the HPA "scalingActive is false" and thrwon with error 
<img width="790" alt="image" src="https://github.com/kedacore/keda/assets/165902584/9ae14d10-c4e7-4666-8f8b-8a1c54db04cf">
